mono/metadata/debug-mono-ppdb.c

```diff
@@ -436,7 +435,8 @@ mono_ppdb_lookup_location (MonoDebugMethodInfo *minfo, uint32_t offset)
 	}
 
 	location = g_new0 (MonoDebugSourceLocation, 1);
-	location->source_file = docname;
+	if (strlen(docname) != 0)
```

Comment:
`strlen` is very wasteful.
Just check first character.
`if (docname [0])`

Reply:
Thanks